addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/FullPathDisclosureAlertPassiveScanRule.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.htmlparser.jericho.Source;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class FullPathDisclosureAlertPassiveScanRule extends PluginPassiveScanner {
+    private final int pluginID = 110009;
+    private static final String MESSAGE_PREFIX = "pscanalpha.fullpathdisclosurealert.";
+
+    @Override
+    public int getPluginId() {
+        return this.pluginID;
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        final int statusCode = msg.getResponseHeader().getStatusCode();
+        final String responseBody = msg.getResponseBody().toString();
+        String evidence;
+        // Matches the presence of one of the words in the response body
+        Pattern errorPattern = Pattern.compile("(?i)error|parse|fatal|warning|notice");
+        Matcher errorMatcher = errorPattern.matcher(responseBody);
+        // matches the presence of either windows style or unix style full paths
+        // example for windows : C:\folder\folder
+        // example for unix : /dir/dir
+        Pattern pathPattern =
+                Pattern.compile("(?i)(/([a-z0-9]+/)+)|(([a-z0-9]:\\\\)+([a-z0-9]+\\\\)+)");
+        Matcher pathMatcher = pathPattern.matcher(responseBody);
+        if (!(statusCode < 200 || statusCode > 299)) return;
+        if (!(errorMatcher.find())) return;
+        if (!pathMatcher.find()) {
+            evidence = "Found word/s : " + errorMatcher.group();
+            buildAlert(evidence, false).raise();
+        } else {
+            evidence = "Found Possible Full Path/s : " + pathMatcher.group();
```

Comment:
Evidence should be only what was actually found in order to be highlighted.

Reply:
Should I remove the word 'possible'?

Reply:
You should just use the matched group. The literal string that you're concatenating won't be part of the text to match.